go-controller/pkg/ovn/udn_isolation.go

```diff
@@ -415,3 +432,66 @@ func (bnc *BaseNetworkController) deleteAdvertisedNetworkIsolation(nodeName stri
 	_, err = libovsdbops.TransactAndCheck(bnc.nbClient, ops)
 	return err
 }
+
+func (oc *DefaultNetworkController) syncUDNIsolation() error {
+	// Find ACLs with old "secondary" naming IDs, update them
+	type aclUpdate struct {
+		old *libovsdbops.DbObjectIDs
+		new *libovsdbops.DbObjectIDs
+	}
+	updates := []*aclUpdate{
+		{oc.getUDNACLDbIDs(denySecondaryACL, libovsdbutil.ACLEgress), oc.getUDNACLDbIDs(denyPrimaryUDNACL, libovsdbutil.ACLEgress)},
+		{oc.getUDNACLDbIDs(legacyAllowHostARPACL, libovsdbutil.ACLEgress), oc.getUDNACLDbIDs(allowHostARPACL, libovsdbutil.ACLEgress)},
+		{oc.getUDNACLDbIDs(denySecondaryACL, libovsdbutil.ACLIngress), oc.getUDNACLDbIDs(denyPrimaryUDNACL, libovsdbutil.ACLIngress)},
+		{oc.getUDNACLDbIDs(legacyAllowHostARPACL, libovsdbutil.ACLIngress), oc.getUDNACLDbIDs(allowHostARPACL, libovsdbutil.ACLIngress)},
+		{oc.getUDNACLDbIDs(allowHostSecondaryACL, libovsdbutil.ACLIngress), oc.getUDNACLDbIDs(allowHostPrimaryUDNACL, libovsdbutil.ACLIngress)},
+	}
+
+	aclsToUpdate := make([]*nbdb.ACL, 0)
+	for _, update := range updates {
+		legacyACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, libovsdbops.GetPredicate[*nbdb.ACL](update.old, nil))
+		if err != nil {
+			return fmt.Errorf("unable to find ACLs for UDN Isolation sync: %w", err)
+		}
+		for _, acl := range legacyACLs {
+			externalIDs := update.new.GetExternalIDs()
+			acl.ExternalIDs = externalIDs
+			aclsToUpdate = append(aclsToUpdate, acl)
+		}
+	}
+	if len(aclsToUpdate) > 0 {
+		ops, err := libovsdbops.CreateOrUpdateACLsOps(oc.nbClient, nil, oc.GetSamplingConfig(), aclsToUpdate...)
```

Comment:
use batching here like https://github.com/ovn-kubernetes/ovn-kubernetes/blob/e0e38b221fad23ec7dd799c1d9bf52723df6dfb8/go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync.go#L133-L135
You can steal batchSize from that file too